DataProcessing/EstimizeEstimateDataDownloader.cs

```diff
@@ -44,148 +42,110 @@ public EstimizeEstimateDataDownloader(string destinationFolder, IMapFileProvider
-
             Directory.CreateDirectory(_destinationFolder);
         }
 
         /// <summary>
         /// Runs the instance of the object.
         /// </summary>
         /// <returns>True if process all downloads successfully</returns>
-        public override bool Run()
+        public override bool Run(DateTime date)
         {
             var stopwatch = Stopwatch.StartNew();
 
             try
             {
-                var companies = GetCompanies().Result.DistinctBy(x => x.Ticker).ToList();
-                var count = companies.Count;
-                var currentPercent = 0.05;
-                var percent = 0.05;
-                var i = 0;
-
-                Log.Trace($"EstimizeEstimateDataDownloader.Run(): Start processing {count.ToStringInvariant()} companies");
+                Log.Trace($"EstimizeEstimateDataDownloader.Run(): Start processing");
 
                 var tasks = new List<Task>();
-
-                foreach (var company in companies)
-                {
-                    // Include tickers that are "defunct".
-                    // Remove the tag because it cannot be part of the API endpoint.
-                    // This is separate from the NormalizeTicker(...) method since
-                    // we don't convert tickers with `-`s into the format we can successfully
-                    // index mapfiles with.
-                    var estimizeTicker = company.Ticker;
-                    string ticker;
-
-                    if (!TryNormalizeDefunctTicker(estimizeTicker, out ticker))
-                    {
-                        Log.Error($"EstimizeEstimateDataDownloader(): Defunct ticker {estimizeTicker} is unable to be parsed. Continuing...");
-                        continue;
-                    }
-
-                    if (_processTickers != null && !_processTickers.Contains(ticker, StringComparer.InvariantCultureIgnoreCase))
-                    {
-                        Log.Trace($"EstimizeEstimateDataDownloader.Run(): Skipping {ticker} since it is not in the list of predefined tickers");
-                        continue;
-                    }
-
-                    // Begin processing ticker with a normalized value
-                    Log.Trace($"EstimizeEstimateDataDownloader.Run(): Processing {ticker}");
-
-                    // Makes sure we don't overrun Estimize rate limits accidentally
-                    IndexGate.WaitToProceed();
-
-                    tasks.Add(
-                        HttpRequester($"/companies/{ticker}/estimates")
-                            .ContinueWith(
-                                y =>
+                // Makes sure we don't overrun Estimize rate limits accidentally
+                IndexGate.WaitToProceed();
+
+                tasks.Add(
+                    // An extra day for redundancy
+                    HttpRequester($"/estimates?start_date={date.AddDays(-2):yyyy-MM-dd}&end_date={date:yyyy-MM-dd}")
```

Comment:
`-2` will fall on a weekend. I'd use `-5` since we never have a 5-day holiday+weekend.